### DISCUSSION TITLE
How do I set the title of the mouseover text in the dock? (macos)

### DISCUSSION THREAD

**Discussion by zakhenry** [tauri-apps/tauri#10689] [Q&A]
<img width="159" alt="Screenshot 2024-08-20 at 9 09 39 AM" src="https://github.com/user-attachments/assets/b0c40b8c-a444-4c61-9534-4027110ce6cc">

Using tauri `2.0.0-rc.3`

I'm launching my app from python via a CFFI shared lib (weird I know), and what I end up with is the app in the dock being named `Python`. The actual application in the top left has the correct name from `productName` in `tauri.conf.json`, but I can't work out where the string `Python` is coming from, or where I can configure it either.

Looking at Activity Monitor I can also see the processes are named `Python` which isn't ideal either

<img width="852" alt="Screenshot 2024-08-20 at 9 13 59 AM" src="https://github.com/user-attachments/assets/3be01049-40ba-4bf6-90f9-6e78bd4868a0">

relevant process highlighted.

---

My initial thought was that the process itself was being named `Python`, but that doesn't seem to be the case as printing the name of the process started gives `Process-1`.

Apologies if this is actually not a tauri question, but it would be good to understand where the app gets it's name from.

**Comment by oknoorap** [comment]
Can we do that, @lucasfernog? I've tried to change 'productName' in tauri.conf.json, but no luck.

**Reply by hrynevychroman** [reply]
I found that we can change this inside `Cargo.toml` file:
<img width="626" alt="image" src="https://github.com/user-attachments/assets/8888b24f-ff59-4c9a-87c5-fee3e9da4a80" />

**Comment by WSH032** [comment]
You can refer to the answer here: <https://github.com/WSH032/pytauri/issues/2#issuecomment-2519992053>

> the actual binary executable is /bin/python. This is why you see the Python field in the menu bar and about modal. To solve this issue, you need to generate your own executable file, so you can set the identifier, icon, about, etc., for your executable.

If you don't want to see the `Python` field, you need to compile an executable file to call `libpython` instead of using python to call `libtauri`. This is how we do it in pytauri: <https://wsh032.github.io/pytauri/0.2/usage/tutorial/build-standalone/>
